pkg/local_object_storage/metabase/select.go

```diff
@@ -51,636 +33,35 @@ func (db *DB) Select(cnr cid.ID, filters object.SearchFilters) ([]oid.Address, e
-				var id oid.ID
-				if err := id.Decode(k); err == nil {
-					appendOID(id)
-				}
-				return nil
-			})
-			if err != nil {
-				db.log.Debug("could not iterate over the buckets",
-					zap.Error(err),
-				)
-			}
-		}
-	}
-}
-
-// matchSlowFilters return true if object header is matched by all slow filters.
-func (db *DB) matchSlowFilters(tx *bbolt.Tx, addr oid.Address, f object.SearchFilters, currEpoch uint64) bool {
-	if len(f) == 0 {
-		return true
-	}
-
-	buf := make([]byte, addressKeySize)
-	obj, err := db.get(tx, addr, buf, true, false, currEpoch)
-	if err != nil {
-		return false
-	}
-
-	for i := range f {
-		op := f[i].Operation()
-		if objectcore.IsIntegerSearchOp(op) {
-			attr := f[i].Header()
-			if attr != object.FilterCreationEpoch && attr != object.FilterPayloadSize {
-				break
-			}
-
-			// filter by creation epoch or payload size, both uint64
-			filterVal := f[i].Value()
-			if len(filterVal) == 0 {
-				return false
-			}
-
-			if filterVal[0] == '-' {
-				if op == object.MatchNumLT || op == object.MatchNumLE {
-					return false
-				}
-				continue
-			}
-
-			if len(filterVal) > 20 { // max uint64 strlen
-				if op == object.MatchNumGT || op == object.MatchNumGE {
-					return false
-				}
-				continue
-			}
-
-			num, err := strconv.ParseUint(filterVal, 10, 64)
-			if err != nil {
-				if errors.Is(err, strconv.ErrRange) {
-					if op == object.MatchNumGT || op == object.MatchNumGE {
-						return false
-					}
-					continue
-				}
-				// has already been checked
-				db.log.Debug("unexpected failure to parse numeric filter uint value", zap.Error(err))
-				return false
-			}
-
-			var objVal uint64
-			if attr == object.FilterPayloadSize {
-				objVal = obj.PayloadSize()
-			} else {
-				objVal = obj.CreationEpoch()
-			}
-
-			switch {
-			case objVal > num:
-				if op == object.MatchNumLT || op == object.MatchNumLE {
-					return false
-				}
-			case objVal == num:
-				if op == object.MatchNumLT || op == object.MatchNumGT {
-					return false
-				}
-			case objVal < num:
-				if op == object.MatchNumGT || op == object.MatchNumGE {
-					return false
-				}
-			}
-
-			continue
-		}
-
-		matchFunc, ok := db.matchers[f[i].Operation()]
-		if !ok {
-			return false
-		}
-
-		var data []byte
-
-		switch f[i].Header() {
-		case object.FilterVersion:
-			data = []byte(obj.Version().String())
-		case object.FilterPayloadHomomorphicHash:
-			cs, _ := obj.PayloadHomomorphicHash()
-			data = cs.Value()
-		case object.FilterCreationEpoch:
-			data = make([]byte, 8)
-			binary.LittleEndian.PutUint64(data, obj.CreationEpoch())
-		case object.FilterPayloadSize:
-			data = make([]byte, 8)
-			binary.LittleEndian.PutUint64(data, obj.PayloadSize())
-		default:
-			continue // ignore unknown search attributes
-		}
-
-		if !matchFunc.matchSlow(f[i].Header(), data, f[i].Value()) {
-			return false
+		if len(newCursor) == 0 {
+			break
 		}
+		cursor = string(newCursor)
```

Comment:
shouldnt it be Base64-encoded? Afaik `PreprocessSearchQuery` expects it in API format